### ISSUE TITLE
Reloadable config

### ISSUE THREAD

**Issue by costinm** [open-telemetry/opentelemetry-specification#3771]
If the config is loaded from a file - or from a K8S ConfigMap - changes to the config should reconfigure the SDK if possible.

Some fields may not be re-configurable - but at least 'disabled' and trace ratio should be.

I don't know what are the plans for the logging config: but if something similar to traces is added to allow sampling and/or 
customization of log level - this should also be part of the minimal set of reconfigurable settings. 

Env variables obviously require restarting the app to change the trace sampling rate - which doesn't work very well.

**Comment by costinm** [comment]
I agree - this is a multi-part problem. The mechanisms to reload - watch file, poll, push - can be complicated, but there is broad use of reloading a file and all we really need in the config spec is to indicate that SDKs should/may support reloading the config, and some indication on which fields of the config are recommended to be re loadable. 

How to do it would clearly be out of scope - and happy to help prototype this, I am working on Istio telemetry and we have our own config ( and mechanism to push the config ) - but I am hoping that we can adopt the SDK config or a subset. We use Envoy XDS - OpAMP is a different approach for pushing config - and I'm sure there will be others.

BTW - it may be worth clarifying what 'SDK configuration' means - it is not clear if it means the 'official' SDKs provided by 
otel org, or any telemetry client implementing the  OTLP protocol ( as an independent SDK ). Envoy is not using one of the
official SDKs - but supports the protocol and Istio ( and others ) may translate from the (standard) config format to Envoy 
internal configs.

**Comment by jack-berg** [comment]
Marking as "needsmoreinfo" because we plan on doing this but its a multi-part problem and we're not ready to focus on it at the moment.
